### PR TITLE
Qute: add JsonEscaper

### LINKED ISSUE #43369: Qute Character Escapes for Json Content
### Description

It'll nice to have the same behavior for Json template for character escapes like the one we have for XML content explained here [https://quarkus.io/guides/qute-reference#character-escapes](https://quarkus.io/guides/qute-reference#character-escapes) allowing users to escape content we want to include, for example, as string in an attribute value, for those values that contain quotes, new line, ... characters.
Despite there is a workaround by creating a @TemplateExtension or @TemplateData to access encode static functions available in Apache Commons Texts and others, but I think this must be a core functionality.

### Implementation ideas

Use the same approach for escaping XML content, may be including in quarkus.qute.escape-content-types property "application/json" joined to the characters escape policy.
Other easy approach will be include in "Built-in Template Extensions" one for Strings, allowing encoding and unencoding options.

### PR DESCRIPTION
**Pull request by mkouba**
- fixes #43369

This pull request supersedes #43900.

### CODE REVIEW

**Comment by gsmet** [comment]
It will require a note in the migration guide right? Because it will break things if people don't want escaping or worked around it already?

**Comment by mkouba** [comment]
> It will require a note in the migration guide right? Because it will break things if people don't want escaping or worked around it already?

Exactly so. Although I don't think there are many users formatting JSON documents with Qute...

File: independent-projects/qute/core/src/main/java/io/quarkus/qute/CharReplacementResultMapper.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+/**
+ * Makes it possible to replace chars from Basic Multilingual Plane (BMP).
+ *
+ * @see Character#isBmpCodePoint(int)
+ */
+abstract class CharReplacementResultMapper implements ResultMapper {
+
+    @Override
+    public String map(Object result, Expression expression) {
+        return escape(result.toString());
```
Comment (gastaldi) [inline independent-projects/qute/core/src/main/java/io/quarkus/qute/CharReplacementResultMapper.java:12]: Any chance for a NPE to happen here?

**Review by gastaldi — APPROVED** [review]
LGTM

**Comment by geoand** [comment]
I rebased to pick up a CI fix

**Comment by mkouba** [comment]
FTR: https://github.com/quarkusio/quarkus/wiki/Migration-Guide-3.18#character-escapes

**Pull request merged**